internal/util/variables.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2024 Humanitec
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"regexp"
+	"strings"
+)
+
+// PrepareEnvVariables replaces the dollar sign ($) by double dollar sign ($$)
+// in slice strings for using in the docker compose file
+func PrepareEnvVariables(arr []string) []string {
+	// create pattern matching string
+	chars := []string{
+		"@", "#", "!", "&", "%", ":", ";", "=", "|", "/", "~", "'", "_",
```

Comment:
Can you explain where this set of characters come from here and why we need to list the characters individually? Can't we use a character range or some exclusion if we are already using \w + \s?

🤔 I was looking at https://github.com/compose-spec/compose-go/blob/main/template/template.go#L38C6-L38C6 which is the substitution mechanism that docker compose uses, and it allows any prefix character and supports replacements like: `${Var1}$Var2${Var3}`

There might be an opportunity to borrow the same regex pattern and use it for escapes? 🤔